config/application.rb

```diff
@@ -73,6 +73,9 @@ class Application < Rails::Application
 
     # Configure sensitive parameters which will be filtered from the log file.
     config.filter_parameters += %i[password encrypted_password password_salt password_confirmation]
+
+    # Enable support for loading via Psych, required by PaperTrail
+    config.active_record.use_yaml_unsafe_load = true
```

Comment:
```suggestion
config.use_yaml_unsafe_load = false
config.yaml_column_permitted_classes = [
  ::ActiveRecord::Type::Time::Value,
  ::ActiveSupport::TimeWithZone,
  ::ActiveSupport::TimeZone,
  ::BigDecimal,
  ::Date,
  ::Symbol,
  ::Time
]
```
https://github.com/paper-trail-gem/paper_trail/blob/master/doc/pt_13_yaml_safe_load.md#to-continue-using-the-yaml-serializer might be worth looking through, so that people upgrading don't have any issues with legacy data (but also no security issues)

not sure if we just need to add a bunch of internal models (Account, Lead, Opportunity etc) as well

Reply:
> not sure if we just need to add a bunch of internal models (Account, Lead, Opportunity etc) as well

Allowing all classes to be serialized/deserialized via Psych isn't great, but you can't specify the model classes in `application.rb` (they haven't been loaded at this point). Naively moving the configuration to an initializer breaks, with some Active Support classes being marked as a disallowed class in specs. 

I'm also a bit worried about serialization of associated objects - I suspect we'd have to declare all possible compositions, not just the base model classes.

Fixing this might be tricky. I'd suggest that we look at securing serialization in a separate ticket as I've no idea how far down the rabbit hole this is going to go. Right now, leaving `use_yaml_unsafe_load` set to `true` is no more unsafe than what is present in the app - and only appears to impact Paper Trail `versions` records that can't be directly manipulated by users.

In a perfect world we wouldn't be using YAML serialization for Paper Trail, but I think JSON-based serialization is only available on Postgres. I see you've already identified this as an issue per #1146.

Reply:
☝️ Actually, scrap that. The proposed change seems to be working without adding models - at least in specs. I'll be more comfortable making this change after I've given the front end a manual test - will look at doing that tomorrow.